Source/UI/NewUI/PaymentForm/View/ExpiryDateView.swift

```diff
@@ -110,22 +166,12 @@ extension ExpiryDateView: TextFieldViewDelegate {
     //check for max length including added spacers which all equal to 5
     guard !string.isEmpty else { return false }
     
-    var originalText = textField.text
     let replacementText = string.replacingOccurrences(of: " ", with: "")
     
     //verify entered text is a numeric value
-    if !CharacterSet(charactersIn: replacementText).isSubset(of: .decimalDigits) { return false }
-    
-    //put `/` space after 2 digit
-    if range.location == 2 {
-      originalText?.append("/")
-      textField.text = originalText
-    }
-    
-    if range.location == dateFormatTextCount - 1 {
-      updateExpiryDate(to: (originalText ?? "") + replacementText)
-      return false
-    }
-    return true
+    guard CharacterSet(charactersIn: replacementText).isSubset(of: .decimalDigits) else { return false }
+    guard let currentDigit = Int(replacementText) else { return false }
+
+    return validateInput(textField, currentDigit: currentDigit, location: range.location, replacementText: replacementText)
```

Comment:
Could we use an SDK configuration property _smart input validation_ to enable / disable this feature?

I think its definitely well intended, but from UX perspective I'd have to question how many times I write a 5 and I get an `05/` showing on the screen from it. Good user experience should be about behaving in the way user expects, even if sometimes that may be _less smart way_. If we'd be able to add some consumer configurable property for this, then merchant can have full control over this and can even A/B test it if they so wish.

Reply:
Having tried on simulator, I do mind it less and less the more I try, but the other UX question is:
> If I want to type month 00, or 13...19, why does my keyboard feel like its not working?

I fully get the _smart_ bit, but from user point of view, I feel like keyboard is broken. I get feedback of the press and yet nothing happens... Instead of disabling input, should we just toggle error message visibility on each input?

Reply:
This is the requirement. @patrick-hoban-cko please advice

Reply:
@alex-ioja-yang-cko has a very fair point, I have to admit I did find it a bit strange that we actually limit what the customer of a merchant is able to type in (even when we know they shouldn't be typing 13 or 19). 

Let's go with Alex's suggestion, and allow someone to type in 13, or 19, but then just show the error message right away when they do it. Otherwise if they hit the 3 or 9 after typing in 1, they might think something is broken.

Reply:
@alex-ioja-yang-cko please try other branch `feature/PIMOB-1274_expiery_date_error_messages` and let me know you feedback.

Reply:
@patrick-hoban-cko may i make this update another PR or all 4 PRs will be blocked?

Reply:
Okay, are you saying that in https://github.com/checkout/frames-ios/pull/192 Alex's suggestion to allow a user to type in 13 or 19 is fixed?

Reply:
@patrick-hoban-cko this is about making our web and mobile SDKs' behaviour more cohesive. If we no longer want this why did we discuss it at length last week?

Reply:
Let's get in a call to discuss @harry-brown-cko @ehab-al-cko @alex-ioja-yang-cko

Reply:
@patrick-hoban-cko https://github.com/checkout/frames-ios/pull/192 is not supporting Alex's suggesting but it's showing error message when there's any invalid input which we have agreed on last week

Reply:
@alex-ioja-yang-cko we're going to merge this and discuss fine tuning the logic in the office together. Logic stays as is for now!